lib/virt_autotest/esxi_utils.pm

```diff
@@ -0,0 +1,117 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Utilities for running ESXi commands
+# Maintainer: Nan Zhang <nan.zhang@suse.com>
+
+package virt_autotest::esxi_utils;
+
+use base Exporter;
+use Exporter;
+
+use base 'opensusebasetest';
+use strict;
+use warnings;
+use testapi;
+use utils;
+
+our @EXPORT = qw(
+  esxi_vm_get_vmid
+  esxi_vm_power_getstate
+  esxi_vm_power_off
+  esxi_vm_power_on
+  esxi_vm_power_reboot
+  esxi_vm_power_reset
+  esxi_vm_power_shutdown
+  esxi_vm_network_binding
+  esxi_vm_public_ip
+  get_host_timestamp
+  disable_vm_time_synchronization
+);
+
+sub esxi_vm_get_vmid {
+    my $vm_name = shift;
+    my $vim_cmd = "vim-cmd vmsvc/getallvms | grep -i $vm_name | awk -F' ' '{print \$1}'";
+    my (undef, $vmid) = console('svirt')->run_cmd($vim_cmd, domain => 'sshVMwareServer', wantarray => 1);
+    chomp($vmid);
+    return $vmid;
+}
+
+sub esxi_vm_power_getstate {
+    my $vmid    = shift;
+    my $vim_cmd = "vim-cmd vmsvc/power.getstate";
+    my (undef, $power_state) = console('svirt')->run_cmd("$vim_cmd $vmid", domain => 'sshVMwareServer', wantarray => 1);
+    return $power_state;
+}
+
+sub esxi_vm_power_off {
+    my $vmid    = shift;
+    my $vim_cmd = "vim-cmd vmsvc/power.off";
+    return console('svirt')->run_cmd("$vim_cmd $vmid", domain => 'sshVMwareServer', wantarray => 1);
+}
+
+sub esxi_vm_power_on {
+    my $vmid    = shift;
+    my $vim_cmd = "vim-cmd vmsvc/power.on";
+    return console('svirt')->run_cmd("$vim_cmd $vmid", domain => 'sshVMwareServer', wantarray => 1);
+}
+
+sub esxi_vm_power_reboot {
+    my $vmid    = shift;
+    my $vim_cmd = "vim-cmd vmsvc/power.reboot";
+    return console('svirt')->run_cmd("$vim_cmd $vmid", domain => 'sshVMwareServer', wantarray => 1);
+}
+
+sub esxi_vm_power_reset {
+    my $vmid    = shift;
+    my $vim_cmd = "vim-cmd vmsvc/power.reset";
+    return console('svirt')->run_cmd("$vim_cmd $vmid", domain => 'sshVMwareServer', wantarray => 1);
+}
+
+sub esxi_vm_power_shutdown {
+    my $vmid    = shift;
+    my $vim_cmd = "vim-cmd vmsvc/power.shutdown";
+    return console('svirt')->run_cmd("$vim_cmd $vmid", domain => 'sshVMwareServer', wantarray => 1);
+}
```

Comment:
What about aggregate all these functions to one ? esxi_vm_power_cycle which takes $vim_cmd or 'shutdown/reset' as arguments ?

Reply:
Thanks, that's a good idea! Already aggregated them into one function, refer to the latest commit.

Reply:
Why is the routine called like that? power cycle means turning it off and on again. Just call it `esxi_vm_power`.